apps/passport-server/src/services/userService.ts

```diff
@@ -178,6 +183,71 @@ export class UserService {
     }
   }
 
+  public async handleOneClickLogin(
+    email: string,
+    code: string,
+    commitment: string,
+    encryptionKey: string,
+    res: Response
+  ): Promise<void> {
+    if (!this.genericIssuanceService) {
+      throw new PCDHTTPError(500, "Generic issuance service not initialized");
+    }
+    const valid =
+      await this.genericIssuanceService.validateEmailAndPretixOrderCode(
+        email,
+        code
+      );
+    if (!valid) {
+      throw new PCDHTTPError(
+        403,
+        "Invalid Zupass link. Please log in through the home page."
+      );
+    }
+    const existingUser = await fetchUserByEmail(this.context.dbPool, email);
+    if (existingUser) {
+      res.status(200).json({
+        isNewUser: false,
+        encryptionKey: existingUser.encryption_key
```

Comment:
maybe i missed this but where is the case that the user exists but we don't have their encryption key (i.e. they set a password) handled?

Reply:
yeah, later down the file and i tested this too

Reply:
also tested locally